src/AggregateFunctions/AggregateFunctionNull.h

```diff
@@ -153,7 +153,9 @@ class AggregateFunctionNullBase : public IAggregateFunctionHelper<Derived>
 
     void deserialize(AggregateDataPtr __restrict place, ReadBuffer & buf, std::optional<size_t> version, Arena * arena) const override
     {
-        bool flag = true;
+        /// When deserializing a bool it might trigger UBSAN if the input is not 0 or 1, so it's better to use a Int8
+        static_assert(sizeof(bool) == sizeof(Int8));
```

Comment:
Similar sanitizer alerts are popping up in other places, so I'm going to change the readBinary(bool) implementation instead